File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int n = nums.length;
        int output = 0;
        for(int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] == nums[j]) {
                    output++;
                }
            }
        }
        return output;
    }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    output = {}
    count = 0

    function factorialize(num) {
        if (num < 0) 
              return -1;
        else if (num == 0) 
            return 1;

          else {
              return (num * factorialize(num - 1));
          }
      }

      function findPairs(num) {
        if(num <= 1) {
            return 0
        }
        return factorialize(num) / (factorialize(2) * (factorialize(num-2)))
      }

    for (const i in nums){
        if (nums[i] in output) {
            output[nums[i]] += 1  
        }
        else {
            output[nums[i]] = 1   
        }    
    }

    for (const i in Object.values(output)) {
        count += findPairs(parseInt(Object.values(output)[i]))
    }

    return count
    
};